### ISSUE TITLE
Support additional authenticators

### ISSUE THREAD

**Issue by yuvipanda** [jupyterhub/the-littlest-jupyterhub#36]
Currently we only support dummy & firstuse authenticator.

I want us to support arbitrary authenticators without having to add custom code for each of them. z2jh requires we do a bunch of extrenous work mapping YAML config to authenticator config for it to work. I want us to not have to do that, somehow.

**Comment by minrk** [comment]
We could rely on standard config.py so users could follow the given authenticator's installation and config documentation and use traitlets `load_subconfig` to load file(s) in a particular location:

```python
import os
import glob

for f in glob.glob('/etc/jupyterhub/*_config.py'):
    load_subconfig(f)
```

(or it could be just one supported config file at /etc/jupyterhub/jupyterhub_config.py)

Then the simple case is that users edit `/etc/jupyterhub/jupyterhub_config.py` and it's loaded as additional config by /srv/tlj/jupyterhub_config.py.

**Comment by yuvipanda** [comment]
Fixed by a combination of #37, #78
